### PR TITLE
Adds EXCEPT and INTERSECT

### PR DESCRIPTION
**Pull request by isaacsanders**
Based off the excellent work of @feymartynov

### CODE REVIEW

File: test/ecto/adapters/postgres_test.exs [hunk 1]
```diff
@@ -417,31 +433,36 @@ defmodule Ecto.Adapters.PostgresTest do
   end
 
   test "interpolated values" do
```
Comment (josevalim) [inline test/ecto/adapters/postgres_test.exs]: We don't need to change this particular test. Testing one of combinations is enough for what it aims to exercise.

File: test/ecto/query/inspect_test.exs [hunk 2]
```diff
@@ -245,33 +254,51 @@ defmodule Ecto.Query.InspectTest do
   end
 
   test "to_string all" do
```
Comment (josevalim) [inline test/ecto/query/inspect_test.exs]: We don't need to change this particular test. Testing one of combinations is enough for what it aims to exercise.

**Comment by josevalim** [comment]
Thanks @isaacsanders! Can you also please add `except_all` and `intersection_all` for completeness?

**Comment by isaacsanders** [comment]
@josevalim I don't think `except_all` and `intersection_all` exist as part of the SQL spec...

File: test/ecto/query_test.exs [hunk 3]
```diff
@@ -99,12 +99,22 @@ defmodule Ecto.QueryTest do
 
   describe "combinations" do
     test "adds union expressions" do
```
[the remark this answers is not in the document]
Comment (isaacsanders) [reply test/ecto/query_test.exs]: done.

**Comment by isaacsanders** [comment]
I think I've responded to all of your feedback.

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Comment by isaacsanders** [comment]
😁

**Pull request merged**